src/adapter/src/catalog.rs

```diff
@@ -4701,8 +4701,13 @@ impl Catalog {
 
             // Push drop commands.
             match entry.item() {
-                CatalogItem::Table(_) | CatalogItem::Source(_) => {
-                    migration_metadata.previous_source_ids.push(id)
+                CatalogItem::Table(_) => migration_metadata.previous_source_ids.push(id),
+                CatalogItem::Source(source) => {
+                    if matches!(source.data_source, DataSourceDesc::Introspection(_)) {
```

Comment:
@moulimukherjee @sploiselle can you confirm if this is the correct way to identify a storage managed collection?